tests/run_tests.py

```diff
@@ -503,6 +505,8 @@ def runTest(engine, testOptions, testName, testReport, testOut):
     if os.path.exists(testingFile) == False:
         driverFile += ''
         # nothing to be done
+    elif testOptions.exceptions == True:
```

Comment:
can you explain the need for this flag? why is the driver file different for the exception tests?

Reply:
The exception tests don't seem to fit in the existing drivers. For example, the commonjs driver has this line at the end:
`require('./exitexception2').then(_ => {onInstantiation(_)})` This triggers the exception outside of the try/catch.

Reply:
you can change it for all cases. 
But maybe it's better to use a dedicated function for handling the error, like `onError(err)`.
for all other tests you can just implement it as `throw err;`

Reply:
or even better:

```
console.log("unexpected exit: FAILURE");
throw err;
```

so it's more likely that the test will register as failed

Reply:
correct. you can add a commit before this, in this same PR, that adds the infrastructure and updates the existing files